core/src/com/unciv/logic/automation/NextTurnAutomation.kt

```diff
@@ -114,6 +114,16 @@ object NextTurnAutomation {
 
     /** allow AI to spend money to purchase city-state friendship, buildings & unit */
     private fun useGold(civInfo: CivilizationInfo) {
+        if (civInfo.getHappiness() > 0) {
+            // Marry all possible city-states
+            for (unique in civInfo.getMatchingUniques("Can spend Gold to annex or puppet a City-State that has been your ally for [] turns.")) {
```

Comment:
This is a "for unique" check that should be replaced by a "if hasUnique" check (added to the civInfo.getHappiness if, above) - if indeed there's a max on 1 marriage per turn